File: browser-use-python/src/browser_use_sdk/v2/resources/files.py
from __future__ import annotations

from ..._core.http import AsyncHttpClient, SyncHttpClient
from ...generated.v2.models import (
    TaskOutputFileResponse,
    UploadFilePresignedUrlResponse,
)


class Files:
    def __init__(self, http: SyncHttpClient) -> None:
        self._http = http

    def session_url(
        self,
        session_id: str,
        *,
        file_name: str,
        content_type: str,
        size_bytes: int,
    ) -> UploadFilePresignedUrlResponse:
        """Generate a presigned URL for uploading files to an agent session."""
        return UploadFilePresignedUrlResponse.model_validate(
            self._http.request(
                "POST",
                f"/files/sessions/{session_id}/presigned-url",
                json={
                    "fileName": file_name,
                    "contentType": content_type,
                    "sizeBytes": size_bytes,
                },
            )
        )

    def browser_url(
        self,
        session_id: str,
        *,
        file_name: str,
        content_type: str,
        size_bytes: int,
    ) -> UploadFilePresignedUrlResponse:
        """Generate a presigned URL for uploading files to a browser session."""
        return UploadFilePresignedUrlResponse.model_validate(
            self._http.request(
                "POST",
                f"/files/browsers/{session_id}/presigned-url",
                json={
                    "fileName": file_name,
                    "contentType": content_type,
                    "sizeBytes": size_bytes,
                },
            )
        )

    def task_output(self, task_id: str, file_id: str) -> TaskOutputFileResponse:
        """Get secure download URL for a task output file."""
        return TaskOutputFileResponse.model_validate(
            self._http.request(
                "GET",
                f"/files/tasks/{task_id}/output-files/{file_id}",
            )
        )


class AsyncFiles:
    def __init__(self, http: AsyncHttpClient) -> None:
        self._http = http

    async def session_url(
        self,
        session_id: str,
        *,
        file_name: str,
        content_type: str,
        size_bytes: int,
    ) -> UploadFilePresignedUrlResponse:
        """Generate a presigned URL for uploading files to an agent session."""
        return UploadFilePresignedUrlResponse.model_validate(
            await self._http.request(
                "POST",
                f"/files/sessions/{session_id}/presigned-url",
                json={
                    "fileName": file_name,
                    "contentType": content_type,
                    "sizeBytes": size_bytes,
                },
            )
        )

    async def browser_url(
        self,
        session_id: str,
        *,
        file_name: str,
        content_type: str,
        size_bytes: int,
    ) -> UploadFilePresignedUrlResponse:
        """Generate a presigned URL for uploading files to a browser session."""
        return UploadFilePresignedUrlResponse.model_validate(
            await self._http.request(
                "POST",
                f"/files/browsers/{session_id}/presigned-url",
                json={
                    "fileName": file_name,
                    "contentType": content_type,
                    "sizeBytes": size_bytes,
                },
            )
        )

    async def task_output(self, task_id: str, file_id: str) -> TaskOutputFileResponse:
        """Get secure download URL for a task output file."""
        return TaskOutputFileResponse.model_validate(
            await self._http.request(
                "GET",
                f"/files/tasks/{task_id}/output-files/{file_id}",
            )
        )

File: browser-use-node/src/v2/resources/files.ts
import type { HttpClient } from "../../core/http.js";
import type { components } from "../../generated/v2/types.js";

type UploadFileRequest = components["schemas"]["UploadFileRequest"];
type UploadFilePresignedUrlResponse = components["schemas"]["UploadFilePresignedUrlResponse"];
type TaskOutputFileResponse = components["schemas"]["TaskOutputFileResponse"];

export class Files {
  constructor(private readonly http: HttpClient) {}

  /** Generate a presigned URL for uploading files to an agent session. */
  sessionUrl(sessionId: string, body: UploadFileRequest): Promise<UploadFilePresignedUrlResponse> {
    return this.http.post<UploadFilePresignedUrlResponse>(
      `/files/sessions/${sessionId}/presigned-url`,
      body,
    );
  }

  /** Generate a presigned URL for uploading files to a browser session. */
  browserUrl(sessionId: string, body: UploadFileRequest): Promise<UploadFilePresignedUrlResponse> {
    return this.http.post<UploadFilePresignedUrlResponse>(
      `/files/browsers/${sessionId}/presigned-url`,
      body,
    );
  }

  /** Get secure download URL for a task output file. */
  taskOutput(taskId: string, fileId: string): Promise<TaskOutputFileResponse> {
    return this.http.get<TaskOutputFileResponse>(
      `/files/tasks/${taskId}/output-files/${fileId}`,
    );
  }
}
